src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/CoinviewRule.cs

```diff
@@ -90,10 +90,13 @@ public override async Task RunAsync(RuleContext context)
                         ConsensusErrors.BadBlockSigOps.Throw();
                     }
 
-                    if (!this.IsProtocolTransaction(tx))
+                    if (!tx.IsCoinBase)
                     {
                         this.CheckInputs(tx, view, index.Height);
-                        fees += view.GetValueIn(tx) - tx.TotalOut;
+
+                        if (!tx.IsCoinStake)
```

Comment:
use IsProtocolTransaction

Reply:
Not  sure why. Surely `if (!tx.IsCoinBase)` and `if (!tx.IsCoinStake)` is equivalent to `!IsProtocolTransaction(tx)` ?

Reply:
Yes they are and exactly for that reason you should use `IsProtocolTransaction()` 😉

Reply:
Given the current use cases this is exactly equivalent. I will make the change though.

Reply:
This is because we then keep and reference to stake in the POS classes, else Bitcoin network will make this check as well.

Reply:
I fully realize that. Currently the variable is not unique to POSTransaction though. It comes from the shared Transaction class..